### DISCUSSION TITLE
Cannot find DLL of built library

### DISCUSSION THREAD

**Discussion by stamminator** [dotnet/runtime#64272] [Q&A] [answered]
I'm trying to build Microsoft.VisualBasic.Core for .NET Standard 2.0 because the .NET Standard version of the Microsoft.VisalBasic nuget package is missing almost all of the classes for some reason (the .NET Framework and .NET Core+ versions have the full implementation). I've followed the directions at https://github.com/dotnet/runtime/blob/main/docs/workflow/building/libraries/README.md. The CoreCLR built just fine, then I ran `.\build.cmd -projects .\src\libraries\Microsoft.VisualBasic.Core\Microsoft.VisualBasic.Core.sln -f netstandard2.0 -c Release`, which built successfully with no warnings or errors.

After that, I figured there'd be a DLL in `artifacts\bin\Microsoft.VisualBasic.Core\netstandard2.0-Release` that I can import into my .NET Standard 2.0 project that I need it for, but the directory is empty. I also tried `dotnet pack .\src\libraries\Microsoft.VisualBasic.Core\src\ -c Release`, but to be honest, I don't understand how a pack differs from a build. This is my first time ever building the runtime, so I must be missing something obvious. Can anyone tell me what I'm doing wrong?

**Comment by huoyaoyuan** [comment, marked as answer]
There had never been .NET Standard support for Microsoft.VisualBasic.Core.
https://github.com/dotnet/runtime/blob/43ceed5514ca93242ff41f9f1ee53fc108c5dc1b/src/libraries/Microsoft.VisualBasic.Core/src/Microsoft.VisualBasic.Core.vbproj#L19
Modifying the project file is a non-trivial work. It requires additional knowledge of the runtime repository organizing.

**Reply by stamminator** [reply]
I see. That's disappointing. So what are the options for teams with VB libraries they need to reuse across .NET Framework and .NET Core+? I don't have any experience with multi-targeted projects, but do you think that would work?

**Reply by huoyaoyuan** [reply]
I viewed the code, and it seems to contain a lot of Windows specific code.

> So what are the options for teams with VB libraries they need to reuse across .NET Framework and .NET Core+?

The SDK automatically references the appropriate library for .NET Framework and .NET Core (high version). There is no available version for .NET Standard, but you can just set your project to `<TargetFrameworks>net6.0;net48</TargetFrameworks>` to share the code.

**Reply by stamminator** [reply]
Thank you. That's exactly the path I went down and it's worked well enough. Especially when I ran into issues with `Microsoft.AspNetCore.Cryptography.KeyDerivation` being referenceable from .NET Standard 2.0 but crashing on runtime no matter what I did, I eventually settled on the multi-targeting. I have nothing but respect for the folks who've managed to make .NET Standard 2.0 work in a production environment, but I'm not one of them.
